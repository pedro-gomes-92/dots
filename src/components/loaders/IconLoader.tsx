import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { Icon, IconProps } from '../miscellaneous/Icon';

export interface IconLoaderProps extends Omit<IconProps, 'name'> {
  name?: IconProps['name'];
}

export const IconLoader: FunctionComponent<IconLoaderProps> = ({
  className,
  name,
  ...rest
}: IconLoaderProps): JSX.Element => {
  return <Icon name={name} className={classnames('is-loader', 'is-relative', className)} {...rest} />;
};

IconLoader.defaultProps = {
  animations: [{ name: 'rotateRight', duration: 'slow', isInfinite: true }],
  name: 'loader',
};
