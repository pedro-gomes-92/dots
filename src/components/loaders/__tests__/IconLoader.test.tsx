import React from 'react';
import { render } from '@testing-library/react';

import { IconLoader } from '../IconLoader';

describe('Given <IconLoader />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<IconLoader />);
      expect(container).toMatchSnapshot();
    });
  });
});
