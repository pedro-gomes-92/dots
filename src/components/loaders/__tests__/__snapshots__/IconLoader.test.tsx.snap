// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <IconLoader /> When initializes Then should match the snapshot 1`] = `
<div>
  <span
    class="icon is-loader is-relative"
  >
    <i
      class="icon-font icon-font-loader"
    />
  </span>
</div>
`;
